### PR TITLE
Issue/4523 comment pn actions context aware

### LINKED ISSUE #4523: Add 'Like/Approve' actions to a Comment Notification in addition to replying.
Comes from #4031 
We only have the "Reply" action. Adding Like / Approve actions to the notification for quick access to these actions.

### PR DESCRIPTION
**Pull request by mzorz**
Fixes #4523

Needs review: @kwonye 

Following discussions and further investigation, we've implemented the pair REPLY/LIKE and REPLY/APPROVE quick actions following this reasoning:
- these are the most used actions, REPLY being the one that drives content creation / keep continued dialogue
- the APPROVE action appears only when the blog setting is to moderate every comment prior to posting
- the LIKE action appears when the comment is automatically published (no moderation needed) and/or if checks can’t be performed ATM of receiving the PN (also this goes in line with `approval` being an implicit action when `like` happens)

This PR makes all checks for the user to see if they can like/moderate and shows the quick actions accordingly.

To test:

### CASE A: for moderated comments
0) log in to the Android app with an admin of a blog
1) on the web, log in to your admin console: My sites -> Settings -> Discussion tab and check `Comment must be manually approved` under the `Before a comment appears` title
2) have some other user make a comment on such blog on the web or other client
3) the Android app should receive a notification that shows actions REPLY/APPROVE
currentformat

4) Tap on APPROVE, this will show you the comment detail and the APPROVE action should be marked in colour (and the comment will be approved)

### CASE B: for non-moderated comments
0) log in to the Android app with an admin of a blog
1) on the web, log in to your admin console: My sites -> Settings -> Discussion tab and leave the following unchecked `Comment must be manually approved` under the `Before a comment appears` title
2) have some other user make a comment on such blog on the web or other client
3) the Android app should receive a notification that shows actions REPLY/LIKE
like

4) Tap on LIKE, this will show you the comment detail and the LIKE action should be marked in colour (and the comment will be liked)

REPLY remains as usual

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java [hunk 1]
```diff
@@ -207,12 +210,46 @@ private void showGroupNotificationForActiveNotificationsMap(int pushId, String n
...
     private void addActionsForCommentNotification(NotificationCompat.Builder builder, String noteId) {
         // Add some actions if this is a comment notification
-        Intent commentReplyIntent = new Intent(this, WPMainActivity.class);
-        commentReplyIntent.putExtra(WPMainActivity.ARG_OPENED_FROM_PUSH, true);
-        commentReplyIntent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK
-                | Intent.FLAG_ACTIVITY_CLEAR_TASK);
-        commentReplyIntent.setAction("android.intent.action.MAIN");
-        commentReplyIntent.addCategory("android.intent.category.LAUNCHER");
+
+        boolean bActionsSet = false;
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: The only prefix we use is `m` for member variables. Please rename this boolean to a question format, so `areActionsSet`

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java [hunk 2]
```diff
@@ -224,6 +261,44 @@ private void addActionsForCommentNotification(NotificationCompat.Builder builder
                 commentReplyPendingIntent);
     }
 
+    private void addCommentLikeActionForCommentNotification(NotificationCompat.Builder builder, String noteId) {
+        // adding comment like action
+        Intent commentLikeIntent = getCommentActionIntent();
+        commentLikeIntent.addCategory("comment-like");
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: please extract "comment-like" to a constant

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java [hunk 3]
```diff
@@ -224,6 +261,44 @@ private void addActionsForCommentNotification(NotificationCompat.Builder builder
...
+        PendingIntent commentLikePendingIntent = PendingIntent.getActivity(this, 0, commentLikeIntent,
+                PendingIntent.FLAG_CANCEL_CURRENT);
+        builder.addAction(R.drawable.ic_action_like, getText(R.string.like),
+                commentLikePendingIntent);
+    }
+
+    private void addCommentApproveActionForCommentNotification(NotificationCompat.Builder builder, String noteId) {
+        // adding comment approve action
+        Intent commentApproveIntent = getCommentActionIntent();
+        commentApproveIntent.addCategory("comment-moderate");
```
Comment (kwonye) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: please extract ""comment-moderate"" to a constant

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java (hunk 2 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: 👍

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java (hunk 3 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: 👍

File: WordPress/src/main/java/org/wordpress/android/GCMMessageService.java (hunk 1 above)
Comment (mzorz) [inline WordPress/src/main/java/org/wordpress/android/GCMMessageService.java]: 👍

**Pull request merged**